recipes/zstd/all/conanfile.py

```diff
@@ -21,13 +22,19 @@ class ZstdConan(ConanFile):
         "shared": [True, False],
         "fPIC": [True, False],
         "threading": [True, False],
+        "build_programs": [True, False],
     }
     default_options = {
         "shared": False,
         "fPIC": True,
         "threading": True,
+        "build_programs": False,
     }
 
+    def validate(self):
+        if is_msvc(self):
+            raise ConanInvalidConfiguration(f"{self.ref} can not be built with shared on Visual Studio and msvc, use mingw.")
+
```

Comment:
```suggestion
    def validate(self):
        if is_msvc(self):
            raise ConanInvalidConfiguration(f"{self.ref} can not be built with shared on Visual Studio and msvc, use mingw.")

```

It's not clear about this new restriction. What's the error, why only fail now?

Reply:
You are correct. Removed this check to allow msvc build!